chemprop/nn/metrics.py

```diff
@@ -63,160 +77,473 @@ def forward(
+
+    def update(
+        self,
+        preds: Tensor,
+        targets: Tensor,
+        mask: Tensor | None = None,
+        weights: Tensor | None = None,
+        *args,
+    ):
         mask = torch.ones_like(targets, dtype=torch.bool) if mask is None else mask
-        return F.f1_score(
-            preds[mask], targets[mask].long(), threshold=self.threshold, task="binary"
-        )
+        weights = torch.ones_like(targets, dtype=torch.float) if weights is None else weights
 
+        if not (0 <= preds.min() and preds.max() <= 1):  # assume logits
+            preds = preds.sigmoid()
 
-@MetricRegistry.register("bce")
-class BCEMetric(BCELoss, Metric):
-    pass
+        TP, FP, TN, FN = self._calc_unreduced_loss(preds, targets.long(), mask, weights, *args)
 
+        self.TP += [TP]
+        self.FP += [FP]
+        self.TN += [TN]
+        self.FN += [FN]
 
-@MetricRegistry.register("ce")
-class CrossEntropyMetric(CrossEntropyLoss, Metric):
-    pass
+    def _calc_unreduced_loss(self, preds, targets, mask, weights, *args) -> Tensor:
+        TP = (targets * preds * weights * mask).sum(0, keepdim=True)
+        FP = ((1 - targets) * preds * weights * mask).sum(0, keepdim=True)
+        TN = ((1 - targets) * (1 - preds) * weights * mask).sum(0, keepdim=True)
+        FN = (targets * (1 - preds) * weights * mask).sum(0, keepdim=True)
+
+        return TP, FP, TN, FN
+
+    def compute(self):
+        TP = torch.cat(self.TP, dim=0).sum(0, keepdim=True)
+        FP = torch.cat(self.FP, dim=0).sum(0, keepdim=True)
+        TN = torch.cat(self.TN, dim=0).sum(0, keepdim=True)
+        FN = torch.cat(self.FN, dim=0).sum(0, keepdim=True)
+
+        MCC = (TP * TN - FP * FN) / ((TP + FP) * (TP + FN) * (TN + FP) * (TN + FN) + 1e-8).sqrt()
+        MCC = MCC * self.task_weights
+        return 1 - MCC.mean()
 
 
 @MetricRegistry.register("binary-mcc")
-class BinaryMCCMetric(BinaryMCCLoss, Metric):
-    pass
+class BinaryMCCMetric(BinaryMCCLoss):
+    def compute(self):
+        return 1 - super().compute()
+
+
+@LossFunctionRegistry.register("multiclass-mcc")
+class MulticlassMCCLoss(ChempropMetric):
+    """Calculate a soft Matthews correlation coefficient ([mccWiki]_) loss for multiclass
+    classification based on the implementataion of [mccSklearn]_
+    References
+    ----------
+    .. [mccWiki] https://en.wikipedia.org/wiki/Phi_coefficient#Multiclass_case
+    .. [mccSklearn] https://scikit-learn.org/stable/modules/generated/sklearn.metrics.matthews_corrcoef.html
+    """
+
+    def __init__(self, task_weights: ArrayLike = 1.0):
+        """
+        Parameters
+        ----------
+        task_weights :  ArrayLike, default=1.0
+            the per-task weights of shape `t` or `1 x t`. Defaults to all tasks having a weight of 1.
+        """
+        super().__init__()
+        task_weights = torch.as_tensor(task_weights, dtype=torch.float).view(1, -1)
+        self.register_buffer("task_weights", task_weights)
```

Comment:
```suggestion
        super().__init__(task_weights)
```
If that makes sense, please change the rest of them